pylint/lint/progress_reporters.py

```diff
@@ -0,0 +1,60 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/pylint-dev/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/pylint-dev/pylint/blob/main/CONTRIBUTORS.txt
+
+from typing import Optional
+
+
+class BaseProgressReporter:
+    """Progress reporter."""
+
+    def __init__(self) -> None:
+        self.item_count = None
+        self.current_count = 0
+
+    def increment(self, filename: str) -> None:
+        """Print the next filename."""
+        self.current_count += 1
+        msg = filename
+        if self.item_count is not None:
+            msg = f"{msg} ({self.current_count} of {self.item_count})"
+        self.print_message(msg)
+
+    def print_message(self, msg: str) -> None:
```

Comment:
Considering the use of `raise NotImplementedError` this should be an `abstractmethod`

Reply:
Yes, this should be `@abc.abstractmethod`.

This tiny class hierarchy is just to provide a "do nothing" logger, so the code doesn't have "if self.verbose: do some logging", which I've always found cluttering.

Reply:
Pushed, that method is now private as well.

Reply:
Now out-of-date with the removal of the class hierarchy, resolving.